### PR TITLE
api Integration

### PR DESCRIPTION
**Pull request by nishanthmahasamudram**

### CODE REVIEW

File: server/data/assessForEarlyReleaseApiClient.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import { OffenderSummary } from '../@types/assessForEarlyReleaseApiClientTypes'
```
Comment (andrewrlee) [inline server/data/assessForEarlyReleaseApiClient.ts:1]: **nit:** super trivial but might be good to get into the habit of `import type { OffenderSummary}` or `import { type OffenderSummary }` (when importing some types  and other non types). 

Then it removes the possibility of triggering runtime side effects of importing a module

File: server/utils/utils.ts [hunk 2]
```diff
@@ -26,4 +27,12 @@ const initialiseName = (fullName?: string): string | null => {
 
 const hasRole = (user: Express.User, role: AuthRole): boolean => user?.userRoles.includes(role) || false
 
-export { convertToTitleCase, initialiseName, hasRole }
+const parseDate = (date: string) => {
```
Comment (andrewrlee) [inline server/utils/utils.ts:30]: **issue**: we should be receiving iso dates YYYY-MM-DD.
better name `parseIsoDate`  ?

File: server/routes/caseAdmin/caseload.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+      req?.middleware?.clientToken,
+      'MDI',
+    )
+
+    const caseload = offenderSummaryList.map(offender => {
+      return {
+        name: convertToTitleCase(`${offender.firstName} ${offender.lastName}`.trim()),
+        prisonerNumber: offender.prisonerNumber,
+        hdced: offender?.hdced,
+        remainingDays: differenceInDays(parseDate(offender?.hdced), new Date()),
```
Comment (andrewrlee) [inline server/routes/caseAdmin/caseload.ts:20]: Think this is fine for now but ideally we'd parse dates in the client...

File: server/utils/utils.ts (hunk 2 above)
Comment (nishanthmahasamudram) [reply server/utils/utils.ts:30]: we are receiving this [dd/MM/yyyy](https://github.com/ministryofjustice/hmpps-assess-for-early-release-api/pull/14/files#diff-84a4d63fbf1fda5ae236e5c9cfae37374ec57be3d019829f9d97cfd2b2f85c4fR21) from backend API.
Ya will rename it.

Comment (andrewrlee) [reply server/utils/utils.ts]: we shouldn't be - we can correct that

**Review by andrewrlee — APPROVED** [review]
LGTM

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**